### PR TITLE
{CI} Add enable traceback

### PR DESCRIPTION
**Pull request by wangzelin007**
**Related command**

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
CI

**Comment by wangzelin007** [comment]
<div style="width: 120em"> </div>

| a | b | d |
|---|---|---|
| 1 | <div style="width: 120em">self&nbsp;=&nbsp;<azure.cli.testsdk.base.ExecutionResult&nbsp;object&nbsp;at&nbsp;0x7f4024ca35e0><br>cli_ctx&nbsp;=&nbsp;<azure.cli.core.mock.DummyCli&nbsp;object&nbsp;at&nbsp;0x7f40262d5ae0><br>command&nbsp;=&nbsp;'keyvault&nbsp;certificate&nbsp;create&nbsp;--vault-name&nbsp;cli-test-kv-ct-sd-000002&nbsp;-n&nbsp;cert1&nbsp;-p&nbsp;@"/mnt/vss/_work/1/s/src/azure-cli/azure/cli/command_modules/keyvault/tests/latest/policy.json"'<br>expect_failure&nbsp;=&nbsp;False<br><br>&nbsp;&nbsp;&nbsp;&nbsp;def&nbsp;_in_process_execute(self,&nbsp;cli_ctx,&nbsp;command,&nbsp;expect_failure=False):<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;from&nbsp;io&nbsp;import&nbsp;StringIO<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;from&nbsp;vcr.errors&nbsp;import&nbsp;CannotOverwriteExistingCassetteException<br>&nbsp;&nbsp;&nbsp;&nbsp;<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;if&nbsp;command.startswith('az&nbsp;'):<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;command&nbsp;=&nbsp;command[3:]<br>&nbsp;&nbsp;&nbsp;&nbsp;<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;stdout_buf&nbsp;=&nbsp;StringIO()<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;logging_buf&nbsp;=&nbsp;StringIO()<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;try:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;#&nbsp;issue:&nbsp;stderr&nbsp;cannot&nbsp;be&nbsp;redirect&nbsp;in&nbsp;this&nbsp;form,&nbsp;as&nbsp;a&nbsp;result&nbsp;some&nbsp;failure&nbsp;information<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;#&nbsp;is&nbsp;lost&nbsp;when&nbsp;command&nbsp;fails.<br>>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;self.exit_code&nbsp;=&nbsp;cli_ctx.invoke(shlex.split(command),&nbsp;out_file=stdout_buf)&nbsp;or&nbsp;0<br><br>src/azure-cli-testsdk/azure/cli/testsdk/base.py:301:&nbsp;<br>_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;<br>env/lib/python3.10/site-packages/knack/cli.py:245:&nbsp;in&nbsp;invoke<br>&nbsp;&nbsp;&nbsp;&nbsp;exit_code&nbsp;=&nbsp;self.exception_handler(ex)<br>src/azure-cli-core/azure/cli/core/__init__.py:127:&nbsp;in&nbsp;exception_handler<br>&nbsp;&nbsp;&nbsp;&nbsp;return&nbsp;handle_exception(ex)<br>_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;<br><br>ex&nbsp;=&nbsp;CLIError(CannotOverwriteExistingCassetteException("Can't&nbsp;overwrite&nbsp;existing&nbsp;cassette&nbsp;('/mnt/vss/_work/1/s/src/azure-cl...chers&nbsp;succeeded&nbsp;:&nbsp;['method',&nbsp;'scheme',&nbsp;'host',&nbsp;'port',&nbsp;'path',&nbsp;'_custom_request_query_matcher']\nMatchers&nbsp;failed&nbsp;:\n"))<br>args&nbsp;=&nbsp;(),&nbsp;kwargs&nbsp;=&nbsp;{}<br><br>&nbsp;&nbsp;&nbsp;&nbsp;def&nbsp;_handle_main_exception(ex,&nbsp;*args,&nbsp;**kwargs):&nbsp;&nbsp;#&nbsp;pylint:&nbsp;disable=unused-argument<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;if&nbsp;isinstance(ex,&nbsp;CannotOverwriteExistingCassetteException):<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;#&nbsp;This&nbsp;exception&nbsp;usually&nbsp;caused&nbsp;by&nbsp;a&nbsp;no&nbsp;match&nbsp;HTTP&nbsp;request.&nbsp;This&nbsp;is&nbsp;a&nbsp;product&nbsp;error<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;#&nbsp;that&nbsp;is&nbsp;caused&nbsp;by&nbsp;change&nbsp;of&nbsp;SDK&nbsp;invocation.<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;raise&nbsp;ex<br>&nbsp;&nbsp;&nbsp;&nbsp;<br>>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;raise&nbsp;CliExecutionError(ex)<br>E&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;azure.cli.testsdk.exceptions.CliExecutionError:&nbsp;The&nbsp;CLI&nbsp;throws&nbsp;exception&nbsp;CLIError&nbsp;during&nbsp;execution&nbsp;and&nbsp;fails&nbsp;the&nbsp;command.<br><br>src/azure-cli-testsdk/azure/cli/testsdk/patches.py:35:&nbsp;CliExecutionError<br><br>During&nbsp;handling&nbsp;of&nbsp;the&nbsp;above&nbsp;exception,&nbsp;another&nbsp;exception&nbsp;occurred:<br><br>self&nbsp;=&nbsp;<azure.cli.command_modules.keyvault.tests.latest.test_keyvault_commands.KeyVaultCertificateRestoreScenarioTest&nbsp;testMethod=test_keyvault_certificate_soft_delete><br>resource_group&nbsp;=&nbsp;'cli_test_kv_cert_soft_delete_000001'<br>key_vault&nbsp;=&nbsp;'cli-test-kv-ct-sd-000002'<br><br>&nbsp;&nbsp;&nbsp;&nbsp;@ResourceGroupPreparer(name_prefix='cli_test_kv_cert_soft_delete_')<br>&nbsp;&nbsp;&nbsp;&nbsp;@KeyVaultPreparer(name_prefix='cli-test-kv-ct-sd-')<br>&nbsp;&nbsp;&nbsp;&nbsp;def&nbsp;test_keyvault_certificate_soft_delete(self,&nbsp;resource_group,&nbsp;key_vault):<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;self.kwargs.update({<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;'loc':&nbsp;'eastus',<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;'policy_path'&nbsp;:&nbsp;os.path.join(TEST_DIR,&nbsp;'policy.json')<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;})<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;self.cmd('keyvault&nbsp;show&nbsp;-n&nbsp;{kv}',&nbsp;checks=self.check('properties.enableSoftDelete',&nbsp;True))<br>&nbsp;&nbsp;&nbsp;&nbsp;<br>>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;self.cmd('keyvault&nbsp;certificate&nbsp;create&nbsp;--vault-name&nbsp;{kv}&nbsp;-n&nbsp;cert1&nbsp;-p&nbsp;@"{policy_path}"',&nbsp;checks=[<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;self.check('status',&nbsp;'completed'),<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;self.check('name',&nbsp;'cert1')<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;])<br><br>src/azure-cli/azure/cli/command_modules/keyvault/tests/latest/test_keyvault_commands.py:1883:&nbsp;<br>_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;<br>src/azure-cli-testsdk/azure/cli/testsdk/base.py:175:&nbsp;in&nbsp;cmd<br>&nbsp;&nbsp;&nbsp;&nbsp;return&nbsp;execute(self.cli_ctx,&nbsp;command,&nbsp;expect_failure=expect_failure).assert_with_checks(checks)<br>src/azure-cli-testsdk/azure/cli/testsdk/base.py:250:&nbsp;in&nbsp;__init__<br>&nbsp;&nbsp;&nbsp;&nbsp;self._in_process_execute(cli_ctx,&nbsp;command,&nbsp;expect_failure=expect_failure)<br>src/azure-cli-testsdk/azure/cli/testsdk/base.py:313:&nbsp;in&nbsp;_in_process_execute<br>&nbsp;&nbsp;&nbsp;&nbsp;raise&nbsp;ex.exception<br>env/lib/python3.10/site-packages/knack/cli.py:233:&nbsp;in&nbsp;invoke<br>&nbsp;&nbsp;&nbsp;&nbsp;cmd_result&nbsp;=&nbsp;self.invocation.execute(args)<br>src/azure-cli-core/azure/cli/core/commands/__init__.py:663:&nbsp;in&nbsp;execute<br>&nbsp;&nbsp;&nbsp;&nbsp;raise&nbsp;ex<br>src/azure-cli-core/azure/cli/core/commands/__init__.py:726:&nbsp;in&nbsp;_run_jobs_serially<br>&nbsp;&nbsp;&nbsp;&nbsp;results.append(self._run_job(expanded_arg,&nbsp;cmd_copy))<br>src/azure-cli-core/azure/cli/core/commands/__init__.py:697:&nbsp;in&nbsp;_run_job<br>&nbsp;&nbsp;&nbsp;&nbsp;result&nbsp;=&nbsp;cmd_copy(params)<br>src/azure-cli-core/azure/cli/core/commands/__init__.py:333:&nbsp;in&nbsp;__call__<br>&nbsp;&nbsp;&nbsp;&nbsp;return&nbsp;self.handler(*args,&nbsp;**kwargs)<br>src/azure-cli/azure/cli/command_modules/keyvault/_command_type.py:138:&nbsp;in&nbsp;keyvault_command_handler<br>&nbsp;&nbsp;&nbsp;&nbsp;return&nbsp;keyvault_exception_handler(self.command_loader,&nbsp;ex)<br>_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;_&nbsp;<br><br>cmd&nbsp;=&nbsp;<azure.cli.command_modules.keyvault.KeyVaultCommandsLoader&nbsp;object&nbsp;at&nbsp;0x7f4024ca2440><br>ex&nbsp;=&nbsp;CannotOverwriteExistingCassetteException("Can't&nbsp;overwrite&nbsp;existing&nbsp;cassette&nbsp;('/mnt/vss/_work/1/s/src/azure-cli/azure/c...tchers&nbsp;succeeded&nbsp;:&nbsp;['method',&nbsp;'scheme',&nbsp;'host',&nbsp;'port',&nbsp;'path',&nbsp;'_custom_request_query_matcher']\nMatchers&nbsp;failed&nbsp;:\n")<br><br>&nbsp;&nbsp;&nbsp;&nbsp;def&nbsp;keyvault_exception_handler(cmd,&nbsp;ex):<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;from&nbsp;msrest.exceptions&nbsp;import&nbsp;ValidationError,&nbsp;ClientRequestError<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;from&nbsp;azure.cli.core.profiles&nbsp;import&nbsp;ResourceType<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;KeyVaultErrorException&nbsp;=&nbsp;cmd.get_models('KeyVaultErrorException',&nbsp;resource_type=ResourceType.DATA_KEYVAULT)<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;if&nbsp;isinstance(ex,&nbsp;(ValidationError,&nbsp;KeyVaultErrorException)):<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;try:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;raise&nbsp;CLIError(ex.inner_exception.error.message)<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;except&nbsp;AttributeError:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;raise&nbsp;CLIError(ex)<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;elif&nbsp;isinstance(ex,&nbsp;ClientRequestError):<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;if&nbsp;'Failed&nbsp;to&nbsp;establish&nbsp;a&nbsp;new&nbsp;connection'&nbsp;in&nbsp;str(ex.inner_exception):<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;instance_type&nbsp;=&nbsp;'Vault'<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;if&nbsp;'managedhsm'&nbsp;in&nbsp;str(ex.inner_exception):<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;instance_type&nbsp;=&nbsp;'HSM'<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;raise&nbsp;CLIError('Max&nbsp;retries&nbsp;exceeded&nbsp;attempting&nbsp;to&nbsp;connect&nbsp;to&nbsp;{instance_type}.&nbsp;'<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;'The&nbsp;{instance_type}&nbsp;may&nbsp;not&nbsp;exist&nbsp;or&nbsp;you&nbsp;may&nbsp;need&nbsp;to&nbsp;flush&nbsp;your&nbsp;DNS&nbsp;cache&nbsp;'<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;'and&nbsp;try&nbsp;again&nbsp;later.'.format(instance_type=instance_type))<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;raise&nbsp;CLIError(ex)<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;else:<br>>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;raise&nbsp;CLIError(ex)<br>E&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;knack.util.CLIError:&nbsp;Can't&nbsp;overwrite&nbsp;existing&nbsp;cassette&nbsp;('/mnt/vss/_work/1/s/src/azure-cli/azure/cli/command_modules/keyvault/tests/latest/recordings/test_keyvault_certificate_soft_delete.yaml')&nbsp;in&nbsp;your&nbsp;current&nbsp;record&nbsp;mode&nbsp;('once').<br>E&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;No&nbsp;match&nbsp;for&nbsp;the&nbsp;request&nbsp;(<Request&nbsp;(GET)&nbsp;https://cli-test-kv-ct-sd-000002.vault.azure.net/certificates/cert1/pending?api-version=7.4>)&nbsp;was&nbsp;found.<br>E&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Found&nbsp;2&nbsp;similar&nbsp;requests&nbsp;with&nbsp;0&nbsp;different&nbsp;matcher(s)&nbsp;:<br>E&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<br>E&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;1&nbsp;-&nbsp;(<Request&nbsp;(GET)&nbsp;https://cli-test-kv-ct-sd-000002.vault.azure.net/certificates/cert1/pending?api-version=7.4>).<br>E&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Matchers&nbsp;succeeded&nbsp;:&nbsp;['method',&nbsp;'scheme',&nbsp;'host',&nbsp;'port',&nbsp;'path',&nbsp;'_custom_request_query_matcher']<br>E&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Matchers&nbsp;failed&nbsp;:<br>E&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<br>E&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;2&nbsp;-&nbsp;(<Request&nbsp;(GET)&nbsp;https://cli-test-kv-ct-sd-000002.vault.azure.net/certificates/cert1/pending?api-version=7.4>).<br>E&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Matchers&nbsp;succeeded&nbsp;:&nbsp;['method',&nbsp;'scheme',&nbsp;'host',&nbsp;'port',&nbsp;'path',&nbsp;'_custom_request_query_matcher']<br>E&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Matchers&nbsp;failed&nbsp;:<br><br>src/azure-cli/azure/cli/command_modules/keyvault/_command_type.py:51:&nbsp;CLIError</div>| 3 |

**Review by kairu-ms — DISMISSED** [review]

File: scripts/ci/automation_full_test.py [hunk 1]
```diff
@@ -579,12 +580,16 @@ def run_instance_modules(self, instance_modules):
...
+            if enable_traceback:
+                cmd[-1] = '-o junit_family=xunit1 --durations=10'
             parallel_error_flag = process_test(cmd, azdev_test_result_fp, live_rerun=fix_failure_tests)
             pipeline_result = get_pipeline_result(azdev_test_result_fp, pipeline_result) if enable_pipeline_result else None
         if serial_tests:
             azdev_test_result_fp = os.path.join(azdev_test_result_dir, f"test_results_{python_version}_{profile}_{instance_idx}.serial.xml")
             cmd = ['azdev', 'test', '--no-exitfirst', '--verbose', '--series'] + serial_tests + \
                   ['--profile', f'{profile}', '--xml-path', azdev_test_result_fp, '--pytest-args', '-o junit_family=xunit1 --durations=10 --tb=no']
+            if enable_traceback:
+                cmd[-1] = '-o junit_family=xunit1 --durations=10'
```
Comment (bebound) [inline scripts/ci/automation_full_test.py:592]: Shall we made `--tb=no/auto` be an extra variable in command?  So we don't need to duplicate these two lines.

Comment (wangzelin007) [reply scripts/ci/automation_full_test.py:592]: Something like this?
```
pytest_args = '-o junit_family=xunit1 --durations=10'
if enable_traceback:
    pytest_args += ' --tb=no'
cmd = ['azdev', 'test', '--no-exitfirst', '--verbose', '--series'] + serial_tests + \
      ['--profile', f'{profile}', '--xml-path', azdev_test_result_fp, '--pytest-args', pytest_args]
```

Comment (bebound) [reply scripts/ci/automation_full_test.py:592]: Yes

File: scripts/ci/automation_full_test.py [hunk 2]
```diff
@@ -36,6 +36,7 @@
 pull_request_number = os.environ.get('PULL_REQUEST_NUMBER', None)
 enable_pipeline_result = bool(job_name and python_version)
 unique_job_name = ' '.join([job_name, python_version, profile, str(instance_idx)]) if enable_pipeline_result else None
+enable_traceback = True if os.environ.get('ENABLE_TRACEBACK').lower() == 'true' else False
```
Comment (bebound) [inline scripts/ci/automation_full_test.py:39]: Traceback is not enabled by default?

Comment (wangzelin007) [reply scripts/ci/automation_full_test.py:39]: Good question, this default value can be configured directly in the pipeline.
Whether it's enable or disable by default, we can override it when running manually.
image

**Review by jiasli — APPROVED** [review]

**Comment by jiasli** [comment]
Resolve https://github.com/Azure/azure-cli/pull/26824#issuecomment-1623649355

**Review by jsntcy — APPROVED** [review]

**Pull request merged**